21 add volunteer filters

aricionur opened the pull request:
closes #21

aricionur committed:
Copy filters from website

aricionur committed:
Merge branch 'develop' into 21-add-volunteer-filters

aricionur committed:
Merge branch 'develop' into 21-add-volunteer-filters

aricionur committed:
Add translations, fix copy paste bugs, fetch options for fitlers

aricionur committed:
fetch and update filter options in volunteers comp and fix render looping

aricionur committed:
fix switch button colors

aricionur committed:
Rearrange accompanying filter section, add trans

aricionur committed:
Add serialize and deserialize functions for volunteer filters

aricionur committed:
Merge branch 'develop' into 21-add-volunteer-filters

aricionur committed:
Add languages filter and rearrange serializer to avoid duplicate params

aricionur committed:
Add new color and engagement filter, reflect selected filter under header

aricionur committed:
Add availability filter accodion and days items, new CheckButton component

aricionur committed:
add times and occasional filters

aricionur committed:
Add Preferred Availability deutch trasn

aricionur committed:
add serializer and deseritalize for preferred availability

aricionur committed:
Add new engagement options and translations

aricionur committed:
Refactor filter item object creation

aricionur committed:
Create a reusable type and refactor avaliabilty filter creation

aricionur committed:
Create new filter array from selected filters for listing header filter display

aricionur committed:
move clear all button from filters to header

aricionur committed:
put selected filters into API request, refactor useGetQuery

aricionur committed:
some refactors

aricionur committed:
remove clear all button from filters, add new fuction to create filter item for single filters

aricionur committed:
Add debouncer and handle functions for search input changes

aricionur committed:
Fix and rearrange some UI issues

aricionur committed:
Make Filters component reusable

aricionur committed:
move css values to globals css

arturasmckwcz commented on src/components/Dashboard/Volunteers/Filters/constants.tsx:
```diff
@@ -0,0 +1,58 @@
+import { ByDay, OccasionalType } from "need4deed-sdk";
```
these guys need to be shared in SDK

arturasmckwcz replied:
and please check if you could re-use something, like `TimeSlot`

aricionur replied:
I checked the sdk, there is no similar definition for `TimeSlot`, so created new one.

You're right, I added `TimeSlot` and `FilterKeys` to the `sdk`.

But for  `FilterKeys`. I just defined it for volunteers. after adding it into sdk,  for other listing pages we have to add new keys into it.
 
I hope, it will not be a too big enum definition in future..

aricionur replied:
Ohh you alread added `FilterKeys` into sdk as `QueryParamsKeys` 

I will refactor fe branch code...

aricionur replied:
FIXED.

arturasmckwcz commented on src/components/Dashboard/Volunteers/Filters/constants.tsx:
```diff
@@ -0,0 +1,58 @@
+}
+
+export enum FilterKeys {
+  SEARCH = "search",
```
SEARCH won't go to the API, right?
my suggestion would be:
```
import { FilterKeys as ApiFilterKeys } from "need4deed-sdk";

const fiterKeys = {
  ...ApiFilterKeys,
  SEARCH: "search",
}

type FilterKeys = typeof fiterKeys[keyof fiterKeys];
```

then `FilterKeys` will be `"search" | "engagement | ...`

something like that. WDYT?

aricionur replied:
Nope.

SEARCH will go to the API.

frontend do not have all data.

552020 committed:
feat: implement automatic token refresh with axios interceptors

- Add axios configuration with response interceptor for token refresh
- Handle 401/403 errors by automatically refreshing tokens
- Queue failed requests during token refresh process
- Redirect to login on refresh failure
- Initialize axios interceptors in app layout
- Support HTTP-only cookies for secure token management

arturasmckwcz committed:
Merge branch 'develop' into 21-add-volunteer-filters

arturasmckwcz committed:
fix: manual merging error

arturasmckwcz committed:
fix: init interceptor on client side as well

aricionur committed:
move timeslot enum to sdk. fetch filterkeys and timeslot from sdk

aricionur committed:
Merge branch 'develop' into 21-add-volunteer-filters

aricionur committed:
Merge branch '21-add-volunteer-filters' of https://github.com/need4deed-org/fe into 21-add-volunteer-filters

aricionur committed:
serialize language and district to ids for API request

aricionur committed:
Merge branch 'develop' into 21-add-volunteer-filters

arturasmckwcz reviewed (approved):
thank you!!!
